### PR TITLE
Adding definition of question settings, updating license description …

### PR DESCRIPTION
**Pull request by Graspleric**
This updates the format with two fixes:

- we updated the type of the license description to rich content as it can contain links
- we added the `settings` property to questions to facilitate the export of newly added settings

### CODE REVIEW

File: exercise.schema.json [hunk 1]
```diff
@@ -75,12 +75,51 @@
...
+      "type": "object",
+      "properties": {
+        "max_attempts": {
+          "description": "The number of attempts a user has for answering the question correctly",
+          "type": "number",
+          "default": 3
+        },
+        "attempt_settings": {
+          "description": "Settings which apply to a specific attempt. The index of the array is used as the number of the attempt the setting applies to. If no specific settings are given the value will be the default of the property",
+          "type": "array",
```
Comment (screambeard) [inline exercise.schema.json]: @Graspleric i don't think this is an array. The index is important and does not start at zero. It starts at 1 and in theory you could have an entry like:
```
attempt_settings: {
 1:  { "percentage_of_full_points": 100},
 3:  { "percentage_of_full_points": 25},
}
```

**Review by screambeard — CHANGES_REQUESTED** [review]
The question_settings.attempt_settings is not an array. See my comment

**Comment by Graspleric** [comment]
ah check, it is an object with numeric properties. I have updated the schema to reflect this.

File: exercise.schema.json [hunk 2]
```diff
@@ -75,12 +75,51 @@
...
+        },
+        "attempt_settings": {
+          "description": "Settings which apply to a specific attempt. The index of the array is used as the number of the attempt the setting applies to. If no specific settings are given the value will be the default of the property",
+          "type": "object",
+          "patternProperties" : {
+            "^[0-9]$": {
+              "percentage_of_full_points": {
+                "description": "The percentage with which the maximum number of points for this question is multiplied if the user answers the question correctly in this attempt",
+                "type": "number",
+                "default": 0
```
Comment (screambeard) [inline exercise.schema.json:104]: @Graspleric maybe not the best implementation, but it starts at 1 not zero. Zo patternProperties should be [1-9] and default should be 1.

**Review by screambeard — CHANGES_REQUESTED** [review]

File: exercise.schema.json (hunk 2 above)
Comment (Graspleric) [reply exercise.schema.json:104]: @screambeard Check, I have updated the specification to start at 1 and updated the description to explain this as well. 

The default for the `percentage_of_full_points` should be 0 for all attempts except the first one, correct? That is way the default is 0 on this line, but the default for the `attempt_settings` object has 100 on line 109 below.

**Review by screambeard — APPROVED** [review]

**Pull request merged**